### PR TITLE
Support same reset actions on Managers as on Systems

### LINKED ISSUE #901: Support same reset actions on Managers as on Systems
##### SUMMARY

The redfish_command.py module supports commands for performing power reset operations (power on, off, reboot, etc.) on a Redfish System. Redfish Managers have the same model for power reset operations as Systems. But the redfish_command.py module currently only supports a single reset option for Managers - GracefulRestart.

This feature request is to support the same power reset actions on Managers as on Systems. 
 

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

redfish_command.py

##### ADDITIONAL INFORMATION

Currently, there is no way to issue reset actions like On, ForceOff, GracefulShutdown, and ForceRestart on Redfish Managers. These are actions that BMC operators may need to perform.

Example of plays that can be performed after this feature has been implemented:
```yaml
  - name: Turn manager power off
    community.general.redfish_command:
      category: Manager
      command: PowerForceOff
      resource_id: BMC
```

```yaml
  - name: Turn manager power on
    community.general.redfish_command:
      category: Manager
      command: PowerOn
      resource_id: BMC
```

### PR DESCRIPTION
**Pull request by billdodd**
##### SUMMARY

Support the same power reset actions on Managers as on Systems.

Fixes #901 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

redfish_command.py

##### ADDITIONAL INFORMATION

After the change, a play like this can be created:
```paste below
---
- hosts: myhosts
  connection: local
  name: Turn manager power on
  gather_facts: False

  tasks:

  - name: Turn manager power on
    redfish_command:
      category: Manager
      command: PowerOn
      resource_id: BMC
      baseuri: "{{ baseuri }}"
      username: "{{ username }}"
      password: "{{ password }}"
```

Running it will produce output like this:
```
$ ansible-playbook -i myinventory.yml  playbooks/manager/power_on.yml

PLAY [Turn manager power on] *********************************************************

TASK [Turn manager power on] ***************************************************
ok: [mockup-localstorage]

PLAY RECAP *********************************************************************
mockup-localstorage        : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

```

### CODE REVIEW

**Comment by billdodd** [comment]
@smokes2345 You may want to try out this PR to confirm it allows you to perform the Manager reset actions you are trying to achieve.

File: changelogs/fragments/903-enhance-redfish-manager-reset-actions.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - redfish_command - Support same reset actions on Managers as on Systems (https://github.com/ansible-collections/community.general/issues/901)
```
Comment (felixfontein) [inline changelogs/fragments/903-enhance-redfish-manager-reset-actions.yml:2]:
```suggestion
  - redfish_command - support same reset actions on Managers as on Systems (https://github.com/ansible-collections/community.general/issues/901).
```

**Comment by mraineri** [comment]
shipit

**Comment by felixfontein** [comment]
@billdodd tell me when I should merge.

**Comment by billdodd** [comment]
> @billdodd tell me when I should merge.

@felixfontein Ready to merge. Thanks.

**Comment by felixfontein** [comment]
@billdodd thanks for implementing this!
@mraineri thanks for reviewing this!

**Pull request merged**